cmd/snap/cmd_wait.go

```diff
@@ -65,49 +67,47 @@ func isNoOption(err error) bool {
 	return false
 }
 
-func trueish(vi interface{}) bool {
+// trueish takes an interface{} and returns true if the interface value
+// looks "true". For strings thats if len(string) > 0 for numbers that
+// they are != 0 and for maps/slices/arrays that they have elements.
+//
+// Note that *only* types that the json package decode with the
+// "UseNumber()" options turned on are handled here. If this ever
+// needs to becomes a generic "trueish" helper we need to resurrect
+// the code in 306ba60edfba8d6501060c6f773235d8c994a319 (and add nil
+// to it).
+func trueishForJsonUnmarshalledValues(vi interface{}) (bool, error) {
 	switch v := vi.(type) {
+	// limited to the types that json unmarhal can produce
+	case nil:
+		return false, nil
 	case bool:
-		if v == true {
-			return true
-		}
-	case int:
-		if v > 0 {
-			return true
-		}
-	case int64:
-		if v > 0 {
-			return true
-		}
-	case float32:
-		if v > 0 {
-			return true
-		}
-	case float64:
-		if v > 0 {
-			return true
-		}
+		return v, nil
 	case json.Number:
-		if i, err := v.Int64(); err == nil && i > 0 {
-			return true
+		if i, err := v.Int64(); err == nil {
+			return i != 0, nil
 		}
-		if f, err := v.Float64(); err == nil && f != 0.0 {
-			return true
+		if f, err := v.Float64(); err == nil {
+			return f != 0.0, nil
 		}
 	case string:
-		if v != "" {
-			return true
-		}
-	case []interface{}:
-		if len(v) > 0 {
-			return true
+		return v != "", nil
+	}
+	// arrays/slices/maps
+	typ := reflect.TypeOf(vi)
+	switch typ.Kind() {
+	case reflect.Array, reflect.Slice, reflect.Map:
+		s := reflect.ValueOf(vi)
+		if s.Kind() == reflect.Ptr {
```

Comment:
How can that happen? Shouldn't typ.Kind() be Ptr in that case?

Reply:
Yeah, this was from an earlier incarnation of the code, removed now.